lib/camunda-cloud/features/modeling/behavior/CleanUpBusinessRuleTaskBehavior.js

```diff
@@ -0,0 +1,108 @@
+import {
+  is
+} from 'bpmn-js/lib/util/ModelUtil';
+
+import CommandInterceptor from 'diagram-js/lib/command/CommandInterceptor';
+
+const HIGH_PRIORITY = 5000;
+
+
+/**
+ * Zeebe BPMN behavior for ensuring that a BusinessRuleTask:
+ * 1) Either has a taskDefinition ExtensionElement OR
+ * 2) Or has a calledDecision ExtensionElement
+ * 2.1) If it has a calledDecision ExtensionElement, it shall not have taskHeaders
+ */
+export default class CleanUpBusinessRuleTaskBehavior extends CommandInterceptor {
+  constructor(eventBus, modeling) {
+    super(eventBus);
+
+    /**
+     * Remove zeebe:calledDecision when zeebe:taskDefinition is added
+     */
+    this.postExecute([ 'properties-panel.update-businessobject-list',
+      'element.updateModdleProperties'] , HIGH_PRIORITY, function(context) {
+      const {
+        element,
+        currentObject,
+        objectsToAdd,
+        moddleElement,
+        properties
+      } = context;
+
+      // (1) map properties from both commands
+      const extensionElement = currentObject || moddleElement,
+            newValues = objectsToAdd || (properties && properties.values);
+
+      // (2) check conditions and potentially update
+      if (
+        is(element, 'bpmn:BusinessRuleTask')
+        && extensionElement
+        && is(extensionElement, 'bpmn:ExtensionElements')
+        && extensionElement.get('values').some((ele) => is(ele, 'zeebe:CalledDecision'))
+        && newValues
+        && newValues.some((ele) => is(ele, 'zeebe:TaskDefinition'))
```

Comment:
So, when we add a task header, the called decision doesn't get removed?

Reply:
Good thought.

So this would not be possible via the PropertiesPanel.

From a Zeebe perspective: a BusinessRuleTask is a ServiceTask if it has a taskDefinition. So it would not make sense to add a taskHeader to a BusinessRuleTask when it does not have a taskDefinition. However, if it already have a taskDefinition, it could not have a calledDecision :-)

So I think I would leave it the way it is (also to not over-engineer)

WDYT?

Reply:
Let's keep it as it is and wait for feedback 👍